src/backend/egl/error.rs

```diff
@@ -153,7 +153,28 @@ impl EGLError {
 }
 
 /// Wraps a raw egl call and returns error codes from `eglGetError`, if it fails.
-pub fn wrap_egl_call<R, F: FnOnce() -> R>(call: F) -> Result<R, EGLError> {
+pub fn wrap_egl_call<R: PartialEq, F: FnOnce() -> R>(call: F, err: R) -> Result<R, EGLError> {
```

Comment:
Can you please extend the doc comment with a description about the `err` paremeter and what
it is exactly used for.